HubSpot.NET/Api/Company/HubSpotCompanyApi.cs

```diff
@@ -24,25 +26,17 @@ public HubSpotCompanyApi(IHubSpotClient client)
         /// <param name="entity">The entity</param>
         /// <returns>The created entity (with ID set)</returns>
         /// <exception cref="NotImplementedException"></exception>
-        public T Create<T>(T entity) where T : CompanyHubSpotModel, new()
-        {
-            var path = $"{entity.RouteBasePath}/companies";
-
-            return _client.Execute<T>(path, entity, Method.POST);
-        }
+        public CompanyHubSpotModel Create(CompanyHubSpotModel entity)
+            => _client.Execute<CompanyHubSpotModel,CompanyHubSpotModel>(GetRoute<CompanyHubSpotModel>("companies"), entity, Method.POST);
 
         /// <summary>
         /// Gets a specific company by it's ID
         /// </summary>
         /// <typeparam name="T">Implementation of CompanyHubSpotModel</typeparam>
         /// <param name="companyId">The ID</param>
         /// <returns>The company entity</returns>
-        public T GetById<T>(long companyId) where T : CompanyHubSpotModel, new()
-        {
-            var path =  $"{new T().RouteBasePath}/companies/{companyId}";
-
-            return _client.Execute<T>(path, Method.GET);
-        }
+        public CompanyHubSpotModel GetById(long companyId)
+            => _client.Execute<CompanyHubSpotModel>(GetRoute<CompanyHubSpotModel>("companies", companyId.ToString()));
```

Comment:
Could this not be carried through as a long rather than "tostring" here?

Reply:
I'll need to check again, because I don't think the CompanyAPI supported actual `long` values

Reply:
yeah we should create more strongly typed overloads for the `GetRoute<T>` method

Reply:
I created #58 for this